app/models/supermarket.rb

```diff
@@ -1,3 +1,5 @@
 class Supermarket < ApplicationRecord
   has_many :items
+  has_many :customer_items, through: :items
+  has_many :customers, through: :customer_items
```

Comment:
Excellent!! Nice work leveraging the associations of Item

You can even shortcut and just have `has_many :customers, through: :items`